src/hotspot/share/utilities/ostream.cpp

```diff
@@ -384,8 +404,8 @@ char* stringStream::as_string(bool c_heap) const {
 }
 
 stringStream::~stringStream() {
-  if (buffer_fixed == false && buffer != NULL) {
-    FREE_C_HEAP_ARRAY(char, buffer);
+  if (_is_fixed == false && _buffer != NULL && _buffer != _small_buffer) {
```

Comment:
`_is_fixed == false` => `!_is_fixed`
Is `_buffer == NULL` actually possible now?

Reply:
No, you are right. I'll remove it. free would accept NULL anyway, so there is not even a risk.